src/plots/cartesian/axes.js

```diff
@@ -1136,9 +1147,26 @@ function formatDate(ax, out, hover, extraPrecision) {
             }
         }
     }
-    if(suffix && (!ax._inCalcTicks || (suffix !== ax._prevSuffix))) {
-        tt += suffix;
-        ax._prevSuffix = suffix;
+    if(ax.tickmode === 'array') {
+        // we get extra precision in array mode, but it may be useless, strip it off
+        if(tt === '00:00:00') {
+            tt = suffix;
+            suffix = '';
+        }
+        else {
+            tt = tt.replace(/:00$/, '');
+        }
+    }
+
+    if(suffix) {
+        if(hover) {
+            // hover puts it all on one line, so suffix works best up front
+            tt = suffix + (tt ? ' ' + tt : '');
```

Comment:
Previously, hover text on date axes would *not* get the suffix at all. This changes it for all date axes, not just those with array ticks, to include the suffix (as a prefix... hmm, maybe we should change this name...) which is sometimes more information than you need to interpret it, but it's useful often enough that I think this is the right way to go about it.